### DISCUSSION TITLE
Call vehicle when updating vehicle software to 2023.38.9 version_ When the data interface obtains vehicle driving data, it does not return the longitude and latitude where the vehicle is located

### DISCUSSION THREAD

**Discussion by hardont943** [timdorr/tesla-api#772] [General]
![image](https://github.com/timdorr/tesla-api/assets/52837734/b042dbc7-fea2-44af-baf0-8320e0779516)
Have you also encountered such a situation?

**Comment by Basvg** [comment]
This is due to a change Tesla made in firmware version 2023.38, see [here](https://developer.tesla.com/docs/fleet-api#announcements-amp-api-changelog). Location data is not included by default anymore, you need to specifically request it.

> Vehicles running firmware versions 2023.38+ will not return location information by default in vehicle data. Developers who need that information will need to add "location_data" parameter to the query, see [here](https://developer.tesla.com/docs/fleet-api#vehicle_data). This will result in a location sharing icon to show on the vehicle UI.

The /vehicle_data endpoint needs a query string parameter _endpoints_ with the data you need.

> Can be many of 'charge_state', 'climate_state', 'closures_state', 'drive_state', 'gui_settings', 'location_data', 'vehicle_config', 'vehicle_state', 'vehicle_data_combo'

So, for example /vehicle_data?endpoints=drive_state%3Bvehicle_state%3Blocation_data.

**Reply by hardont943** [reply]
thanks！
